codalab/worker/worker_run_state.py

```diff
@@ -264,10 +289,41 @@ def _transition_from_PREPARING(self, run_state):
                     self.dependency_manager.dependencies_dir,
                     self.dependency_manager.get(run_state.bundle.uuid, dep_key).path,
                 )
-                os.symlink(docker_dependency_path, full_child_path)
-            # These are turned into docker volume bindings like:
-            #   dependency_path:docker_dependency_path:ro
-            docker_dependencies.append((dependency_path, docker_dependency_path))
+
+                if dep.child_path != '.':
+                    # A child path can be a nested path which we need to create directories for
+                    Path(full_child_path).mkdir(parents=True, exist_ok=True)
+                    try:
+                        symlink(docker_dependency_path, full_child_path)
+                    except OSError as e:
+                        return run_state._replace(
+                            stage=RunStage.CLEANING_UP, failure_message=str(e)
+                        )
+
+            if dep.child_path == '.':
+                # Process top-level content for the current dependency path
+                for content in os.listdir(dependency_path):
+                    content_child_path = os.path.normpath(
+                        os.path.join(run_state.bundle_path, content)
+                    )
+                    content_dependency_path = os.path.join(dependency_path, content)
+                    docker_dependency_path = os.path.join(docker_dependencies_path, content)
+
+                    # Set up a symlink from the content child path to the container mount location
+                    try:
```

Comment:
Could we try to refactor this a bit - the logic for symlinking/mounting could be a function, so that this part is just looping over the contents of `dependency_path` and calling this one function.